src/main/java/org/terasology/changingBlocks/ChangingBlocksSystem.java

```diff
@@ -63,7 +51,7 @@ public void onSpawn(OnAddedComponent event, EntityRef entity) {
         ChangingBlocksComponent changingBlocks = entity.getComponent(ChangingBlocksComponent.class);
         LocationComponent locComponent = entity.getComponent(LocationComponent.class);
         Block currentBlock = worldprovider.getBlock(locComponent.getWorldPosition(new Vector3f()));
-        String currentBlockFamilyStage = currentBlock.getURI().toString();
+        SimpleUri currentBlockFamilyStage = new SimpleUri(currentBlock.getURI().getModuleName(), currentBlock.getURI().getIdentifier());
```

Comment:
Maybe this should be available on `Block` or `BlockUri` to prevent accidental errors. 🤔 (I'd use at least a module-wide helper class to be sure that we're using the same conversion within the context of Changing Blocks). However, I'm not sure how universal (and generally helpful) this reduction of information is. So, probably the module-wide helper function would indeed be a good compromise.

(We talked about this offline, but just for the record)
The culprit here is that the block uri _can_ contain way more information (shape, orientation, ...) than just the plain block "type" (family?).
Even more confusing, it seems like we're hiding _some_ shapes (if the shape is hard-coded in the block definition maybe?)